packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/OutputTypes/MrwSerializationTypeProvider.cs

```diff
@@ -0,0 +1,387 @@
+            var returnType = typeof(BinaryData);
+            return new CSharpMethod
+            (
+                new MethodSignature(nameof(IPersistableModel<object>.Write), null, null, MethodSignatureModifiers.None, returnType, null, new[] { _serializationOptionsParameter }, ExplicitInterface: _iPersistableModelTInterface),
+                // TO-DO: Call the base model ctor for now until the model properties are serialized https://github.com/microsoft/typespec/issues/3330
+                Return(new NewInstanceExpression(returnType, new ValueExpression[] { new StringLiteralExpression(_iPersistableModelTInterface.Name, false) }))
+            );
+        }
+
+        /// <summary>
+        /// Builds the I model create method.
+        /// </summary>
+        internal CSharpMethod BuildIModelCreateMethod()
+        {
+            Parameter dataParameter = new("data", null, typeof(BinaryData), null, ValidationType.None, null);
+            // IPersistableModel<T>.Create(BinaryData data, ModelReaderWriterOptions options)
+            var typeOfT = GetModelArgumentType(_iPersistableModelTInterface);
+            return new CSharpMethod
+            (
+              new MethodSignature(nameof(IPersistableModel<object>.Create), null, null, MethodSignatureModifiers.None, typeOfT, null, new[] { dataParameter, _serializationOptionsParameter }, ExplicitInterface: _iPersistableModelTInterface),
+              // TO-DO: Call the base model ctor for now until the model properties are serialized https://github.com/microsoft/typespec/issues/3330
+              Return(new NewInstanceExpression(typeOfT, Array.Empty<ValueExpression>()))
+            );
+        }
+
+        /// <summary>
+        /// Builds the I model GetFormatFromOptions method.
+        /// </summary>
+        internal CSharpMethod BuildIModelGetFormatFromOptionsMethod()
+        {
+            ValueExpression jsonWireFormat = SystemSnippets.JsonFormatSerialization;
+            // ModelReaderWriterFormat IPersistableModel<T>.GetFormatFromOptions(ModelReaderWriterOptions options)
+            return new CSharpMethod
+            (
+              new MethodSignature(nameof(IPersistableModel<object>.GetFormatFromOptions), null, null, MethodSignatureModifiers.None, typeof(string), null, new[] { _serializationOptionsParameter }, ExplicitInterface: _iPersistableModelTInterface),
+              jsonWireFormat
+            );
+        }
+
+        /// <summary>
+        /// Builds the serialization constructor for the model.
+        /// </summary>
+        /// <param name="baseinitializationCtor">The base initialization constructor for the model.</param>
+        /// <returns>The constructed serialization constructor. Otherwise <c>null</c> is returned.</returns>
+        internal CSharpMethod? BuildSerializationConstructor(CSharpMethod? baseinitializationCtor)
+        {
+            var baseConstructorParameters = new List<Parameter>();
+            if (baseinitializationCtor != null)
```

Comment:
I don't think we base the serialization ctor off of the init ctor, we shouldn't require that param.

The question we are answering in this method is what would the serialization ctor look like.